packages/babel-plugin-transform-imports/lib/index.js

```diff
@@ -62,61 +62,52 @@ function transform(transformOption, importName, matches) {
-      console.error(
-        `[${transformName}] failed to require transform file ${transformOption}`
-      )
+      throw {
+        err: `[${transformName}] failed to require transform file ${transformOption} ${error}`
+      }
     }
-
     if (typeof transformFn !== 'function') {
-      console.error(
-        `[${transformName}] expected transform function to be exported from ${transformOption}`
-      )
+      throw {
+        err: `[${transformName}] expected transform function to be exported from ${transformOption}`
+      }
     }
-
     let importPath = importName
 
     // Sometimes the import is not located at root level of the src. Examine the source of the specified package and if the designated
     // import is not at the root level, construct a relative path from the source root to it's location.
     if (matches && matches[1]) {
       const packageName = matches[1]
+      const sourceIndex = require.resolve(packageName)
+      const sourceRoot = dirname(sourceIndex)
+
+      const importPaths = globby.sync(
+        [
+          `${sourceRoot}/**/${importName}.js`,
+          `${sourceRoot}/**/${importName}/index.js`
+        ],
+        { cwd: sourceRoot }
+      )
 
-      try {
-        const sourceIndex = require.resolve(packageName)
-        const sourceRoot = dirname(sourceIndex)
-
-        const importPaths = globby.sync(
-          [
-            `${sourceRoot}/**/${importName}.js`,
-            `${sourceRoot}/**/${importName}/index.js`
-          ],
-          { cwd: sourceRoot }
-        )
-
-        if (!importPaths || importPaths.length === 0) {
-          // If there are no import paths found it is the same as if globby or the require.resolve failed, we cannot construct a relative import path
-          // so throw an error and just fall back to using the import name as the path.
-          throw Error()
-        }
-
-        if (importPaths.length > 1) {
-          console.error(
-            `[${transformName}] multiple modules found with the name '${importName}' in '${packageName}'. Continuing using the first match: '${importPaths[0]}'.`
-          )
-        }
-
-        importPath = relative(
-          sourceRoot,
-          importPaths[0].endsWith('index.js')
-            ? dirname(importPaths[0])
-            : importPaths[0]
-        )
-      } catch (error) {
+      if (!importPaths || importPaths.length === 0) {
+        // If there are no import paths found it is the same as if globby or the require.resolve failed, we cannot construct a relative import path
+        // so throw an error and just fall back to using the import name as the path.
         console.error(
           `[${transformName}] no modules match '${importName}' in '${packageName}'. Continuing with '${importName}' as the import path. If that is unexpected or incorrect, make sure '${importName}' exists in '${packageName}' and you have run 'yarn install' to download the package source.`
         )
+        return null
```

Comment:
This is the main fix.
This plugin looks at the imports and replaces them with `${sourceRoot}/**/${importName}.js` or `${sourceRoot}/**/${importName}/index.js`. If we export things and its filename does not exist, it will just display here an error and continue to make a broken build. After this when actually compiling an app (e.g. making a docs bundle) will the error come out.